### PR TITLE
Deployment 10-17-23

### PR DESCRIPTION
**Pull request by daopunk**
Changes:
- create if-else for testnet vs mainnet deployments
- add authorized delegates to testnet deployments

### CODE REVIEW

**Comment by daopunk** [comment]
I didn't create a release, because I only edited the deployment script.

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -82,15 +82,33 @@ abstract contract Deploy is Common, Script {
...
+        // if !delegate: revoke deployer && authorize governor
+        _revokeAllTo(governor);
+      } else if (delegate == deployer) {
+        // if delegate == deployer, maintain deployer auth && authorize governor
+        _delegateAllTo(governor);
+      } else {
+        // if delegate != deployer, authorize delegate && governor
+        _delegateAllTo(delegate);
+        _revokeAllTo(governor);
+      }
```
Comment (pi0neerpat) [inline script/Deploy.s.sol:100]: Are you envisioning that we might have delegates initially on mainnet?

File: script/Deploy.s.sol [hunk 2]
```diff
@@ -246,45 +264,50 @@ contract DeployGoerli is DeployTestnet {
...
-      address(systemCoin), address(collateral[WSTETH]), uint32(ORACLE_INTERVAL_TEST)
-    );
-
-    // deploy denominated oracle of OD/WSTETH denominated against ETH/USD
-    systemCoinOracle = denominatedOracleFactory.deployDenominatedOracle(_odWethOracle, chainlinkEthUSDPriceFeed, false);
+  function setupPostEnvironment() public virtual override updateParams {}
 
-    oracleRelayer.modifyParameters('systemCoinOracle', abi.encode(systemCoinOracle));
-  }
+  /**
```
Comment (pi0neerpat) [inline script/Deploy.s.sol:269]: Do we need to add this back / uncomment?

**Review by pi0neerpat — APPROVED** [review]
Approved with comments

File: script/Deploy.s.sol (hunk 1 above)
Comment (daopunk) [reply script/Deploy.s.sol:100]: that was logic from HAI, but i think the delegate option should actually be removed for mainnet. good point. i'll simplify that before merging this

**Pull request merged**